stingray/utils.py

```diff
@@ -975,3 +986,320 @@ def find_nearest(array, value):
+    ----------
+    final_spectra: object
+        Final Combined AveragedCrossspectrum or AveragedPowerspectrum
+    curr_spec: object
+        AveragedCrossspectrum/AveragedPowerspectrum to be combined
+    flag: bool
+        Indicator variable
+
+    Returns
+    -------
+    object
+        Combined AveragedCrossspectrum/AveragedPowerspectrum
+    """
+    if flag:
+        final_spectra = curr_spec
+        final_spectra.freq = final_spectra.freq.astype('float128')
+        final_spectra.power = final_spectra.power.astype('complex256')
+        final_spectra.unnorm_power = final_spectra.unnorm_power.astype(
+            'complex256')
+
+        return final_spectra
+
+    np.multiply(np.add(final_spectra.freq, curr_spec.freq),
+                curr_spec.m,
+                out=final_spectra.freq)
+    np.multiply(np.add(final_spectra.power, curr_spec.power),
+                curr_spec.m,
+                out=final_spectra.power)
+    np.multiply(np.add(final_spectra.unnorm_power, curr_spec.unnorm_power),
+                curr_spec.m,
+                out=final_spectra.unnorm_power)
+    np.sqrt(np.add(np.square(final_spectra.power_err),
+                   np.square(curr_spec.power_err)),
+            out=final_spectra.power_err)
+
+    final_spectra.m += curr_spec.m
+    final_spectra.df = (final_spectra.df + curr_spec.df) / 2
+    final_spectra.gti = np.concatenate((final_spectra.gti, curr_spec.gti))
+
+    if isinstance(final_spectra, AveragedPowerspectrum):
+        final_spectra.nphots += curr_spec.nphots
+
+    elif isinstance(final_spectra, AveragedCrossspectrum):
+        np.multiply(np.add(final_spectra.pds1.power, curr_spec.pds1.power),
+                    curr_spec.m,
+                    out=final_spectra.pds1.power)
+        np.multiply(np.add(final_spectra.pds2.power, curr_spec.pds2.power),
+                    curr_spec.m,
+                    out=final_spectra.pds1.power)
+        final_spectra.nphots1 += curr_spec.nphots1
+        final_spectra.nphots2 += curr_spec.nphots2
+
+    return final_spectra
+
+
+def _combineSpectra(final_spectra):
+    """
+    Create a final spectra that is the mean of all spectra.
+    Parameters
+    ----------
+    final_spectra: :class:`stingray.AveragedCrossspectrum/AveragedPowerspectrum' object
+        Summed spectra of all spectra
+    Returns
+    -------
+    object
+        Final resulting spectra.
+    """
+    final_spectra.freq /= final_spectra.m
+    final_spectra.power /= final_spectra.m
+    final_spectra.unnorm_power /= final_spectra.m
+    # REVIEW: final_spectra.power_err /= final_spectra.m
+
+    if isinstance(final_spectra, AveragedPowerspectrum):
+        return final_spectra
+
+    elif isinstance(final_spectra, AveragedCrossspectrum):
+        final_spectra.pds1.power /= final_spectra.m
+        final_spectra.pds2.power /= final_spectra.m
+
+        return final_spectra
+
+
+def createChunkedSpectra(spec_type, segment_size, norm, gti, power_type, silent, dt):
```

Comment:
I think this cannot belong here, as `utils` is imported among AveragedCrossspectrum's dependencies. I would move these functions to AveragedCrossspectrum itself

Reply:
All the functions? I was thinking that as well. Should I create a separate file?